modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2063,6 +2069,59 @@ void GDScriptAnalyzer::reduce_array(GDScriptParser::ArrayNode *p_array) {
 	p_array->set_datatype(arr_type);
 }
 
+void GDScriptAnalyzer::update_const_expression_builtin_type(GDScriptParser::ExpressionNode *p_expression, const GDScriptParser::DataType &p_type, const char *p_usage, bool p_is_cast) {
+	if (p_expression->get_datatype() == p_type) {
+		return;
+	}
+	if (p_type.kind != GDScriptParser::DataType::BUILTIN && p_type.kind != GDScriptParser::DataType::ENUM) {
+		return;
+	}
```

Comment:
I'm not sure if a silent return here is correct. It seems to me that either `ERR_FAIL_COND` is needed here, or the method should return `bool`.

Reply:
This is not an error case, if an expected type is not built-in or enum (so a variant or an object) then there is nothing to do - no conversion will take place. There are checks afterwards in all corresponding places that deal with checking that types do match. 

Usage of the method assumes that you call it before storing the resulting type of an expression, like `update_array_literal_element_type` and `reduce_expression`, no need to return booleans.